src/compiler/types.ts

```diff
@@ -643,21 +644,23 @@ namespace ts {
         initializer?: Expression;           // Optional initializer
     }
 
-    export interface ObjectLiteralElement extends Declaration {
+    export interface ObjectLiteralElementLike extends Declaration {
         _objectLiteralBrandBrand: any;
         name?: PropertyName;
    }
 
+    export type ObjectLiteralElement = PropertyAssignment | ShorthandPropertyAssignment | SpreadObjectLiteralAssignment | MethodDeclaration | AccessorDeclaration;

```

Comment:
I don't understand the addition of `ObjectLiteralElementLike` plus this type. Why is it split now?